src/components/timers/Stopwatch.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { useState, useEffect } from 'react';
+
+//Time starts at 0 and not running
+//max time is 2 minutes and 30 seconds
+const Stopwatch: React.FC = () => {
+  const [time, setTime] = useState(0);
+  const [isRunning, setIsRunning] = useState(false);
+  const MAX_TIME = 150000; // 2 minutes and 30 seconds in milliseconds
+
+//intervalID will hold the interval ID
+  useEffect(() => {
+    let intervalId: number;
+    //Check if the stopwatch is running
+    if (isRunning) {
+      // if it is start an interval that runs every 10 milliseconds
+      intervalId = setInterval(() => {
+        setTime(prevTime => {
+          // Stop the timer if it reaches MAX_TIME
+          if (prevTime + 10 >= MAX_TIME) {
+            setIsRunning(false);
+            return MAX_TIME;
+          }
+          return prevTime + 10;
+        });
+      }, 10);
+    }
+    //Return a cleanup function
+    return () => clearInterval(intervalId);
+  }, [isRunning]);
+
+  //Will start and stop the watch
+  const handleStartStop = () => {
+    //if it is true, make it false, if its false make it true
+    setIsRunning(!isRunning);
+  };
+
+  // Resets the stopwatch back to zero and stops it
+  const handleReset = () => {
+    setTime(0);
+    setIsRunning(false);
+  };
+
+  // Sets the stopwatch to the maximum time and stops it
+  const handleEnd = () => {
+    setIsRunning(false);
+    setTime(MAX_TIME); // Set time to max when ending
+  };
+
+  // Function to format time in MM:SS.mmu format
+  const formatTime = (timeInMilliseconds: number): string => {
```

Comment:
these functions should be defined in `utils`, not in the render method of your component